lambda_code/tests/test_mesh.py

```diff
@@ -76,21 +83,69 @@ def setUp(self):
     @mock_s3
     def test_write_to_s3(self):
         output_bucket = create_a_bucket(self.bucket)
-        records = [MeshImmunisationReportEntry("error1"), MeshImmunisationReportEntry("error2"), ]
+        records = [
+            MeshImmunisationReportEntry("error1"),
+            MeshImmunisationReportEntry("error2"),
+        ]
         # TODO: create a method here to read the object. make this to test if write_report is working.
         self.mesh_output.write_report(records)
 
 
-csv_file_name = "data2.csv"
-csv_file_path = f"{os.path.dirname(os.path.abspath(__file__))}/sample_data/{csv_file_name}"
-
-
 class TestMeshCsvParser(unittest.TestCase):
+    """Test the MeshCsvParser class"""
+
     def setUp(self):
-        with open(csv_file_path, "r") as data:
-            self.sample_csv = data.read()
+        self.data_path = f"{os.path.dirname(os.path.abspath(__file__))}/sample_data"
+        self.csv_file_name = "test_mesh_data.csv"
+        self.csv_file_path = f"{self.data_path}/{self.csv_file_name}"
+        self.successful_happy_output_file_name = "successful_happy_output.json"
+        self.successful_happy_output_file_path = (
+            f"{self.data_path}/{self.successful_happy_output_file_name}"
+        )
+        self.failed_happy_output_file_name = "failed_happy_output.json"
+        self.failed_happy_output_file_path = (
+            f"{self.data_path}/{self.failed_happy_output_file_name}"
+        )
+
+        # with open(self.csv_file_path, "r", encoding="utf-8") as data:
+        #    self.sample_csv = data.read()
+
+        self.successful_happy_output_file_path = (
+            f"{self.data_path}/{self.successful_happy_output_file_name}"
+        )
+        with open(
+            self.successful_happy_output_file_path, "r", encoding="utf-8"
+        ) as data:
+            self.successful_happy_output = data.read()
+
+        with open(self.failed_happy_output_file_path, "r", encoding="utf-8") as data:
+            self.failed_happy_output = data.read()
 
     def test_parse_csv(self):
-        mesh_csv = MeshCsvParser(self.sample_csv)
+        """
+        Test that we get a ImmunizationModel object and a ImmunizationErrorModel from parsing
+        the CSV
+        """
+        mesh_csv = MeshCsvParser(self.csv_file_path)
         records = mesh_csv.parse()
         self.assertEqual(len(records), 2)
+
+    def test_successful_immunizations_happy_output(self):
+        """Test that the successful immunization record is formatted correctly"""
+        mesh_csv = MeshCsvParser(self.csv_file_path)
+        immunisation_records, _ = mesh_csv.parse()
+        immunisation_record_json = json.dumps(
+            immunisation_records[0].dict(), indent=4, default=str
+        )
+
+        assert json.loads(immunisation_record_json) == json.loads(
```

Comment:
Please use `unittest` assertions. They'll give you diff and they're integrated with the test-runner. Like `self.assertDictEqual()` .